File: FlowerApp/src/main/java/com/bms/services/ProductService.java
package com.bms.services;

import java.util.HashMap;
import java.util.List;
import java.util.Optional;

import com.bms.entities.Product;

public interface ProductService {

	public Product addProduct(String requestData);

	public Optional<Product> getProduct(String productId);
	
	public List<Product> getAllProducts();

	public HashMap<String, Object> deleteProduct(String productId);

}

File: bmsgit/src/app/product.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Product } from '../app/model/Product';


@Injectable({
  providedIn: 'root'
})
export class ProductService {
  deleteProductD(productId: any) {
    return this.http.post(this.apiUrl+"/deleteProduct",productId);
  }
  apiUrl:string="http://localhost:8084";
  constructor(private http :HttpClient) { }
  getAllProducts(){
    return this.http.get(this.apiUrl+"/getAllProducts")
  }
  saveProduct(product:Product){
    return this.http.post(this.apiUrl+"/addProduct",product)
  }

}
